sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1735,6 +1735,24 @@ object GpuOverrides {
           GpuCustomShuffleReaderExec(childPlans.head.convertIfNeeded(),
             exec.partitionSpecs)
         }
+      }),
+    exec[AdaptiveSparkPlanExec]("Adaptive query", (exec, conf, p, r) =>
+      new SparkPlanMeta[AdaptiveSparkPlanExec](exec, conf, p, r) {
+        override def tagPlanForGpu(): Unit =
+          willNotWorkOnGpu("this is an adaptive plan")
```

Comment:
Do we need to be a bit more descriptive here?  I can see users wondering why it says, `this is an adaptive plan` yet the docs say the plugin support AQE.  "If it supports AQE, why isn't it handling adaptive plans?  Isn't that the whole point of the plugin supporting AQE?" :smile:

Reply:
Yeah, I agree. This whole approach is still confusing. What I really wanted to do was not have any messages related to the AQE operators. Currently, we either replace an operator or show a warning. I think we need a third option of "this does not need replacing". I'm going to look at this again today and see if this can be implemented without major surgery on GpuOverrides.

Reply:
I totally agree.  There are other places we want this too.  Perhaps it is just a list of class names that we should not warn about at all.

Reply:
I pushed an update to introduce a new `DoNotReplaceSparkPlanMeta` rule and logic to suppress warnings about operators that use this rule. I manually tested with TPC-DS with AQE and it looks a lot cleaner now.